CMakeLists.txt

```
@@ -90,11 +90,15 @@ endif()
 
 set(H3_COMPILE_FLAGS "")
 set(H3_LINK_FLAGS "")
+if(WIN32)
+    list(APPEND H3_COMPILE_FLAGS /W2)
+else()
+    list(APPEND H3_COMPILE_FLAGS -Wall)
+endif()
+
```

Maybe prefix this with something like `option(ENABLE_WARNINGS "Enables compiler warnings" ON)`

---

Can we do this in a follow-up PR instead of this PR?

We always use `-Wall` on non Windows even when we don't have this change. So we can treat the new option as a logically separated improvement.

---

```suggestion
option(ENABLE_WARNINGS "Enables compiler warnings" ON)
if(ENABLE_WARNINGS)
    if(WIN32)
        list(APPEND H3_COMPILE_FLAGS /W2)
    else()
        list(APPEND H3_COMPILE_FLAGS -Wall)
    endif()
endif()

```

Here's the code that's necessary. With it being default to on, it should be enabled for all platforms as you're after but it's also possible to get the old behavior back if someone desires it.

---

I've applied your suggestion.